### PR TITLE
Add a note about the offset requirement in moduleEngine struct

### PR DESCRIPTION
**Pull request by anuraaga**
Signed-off-by: Anuraag Agrawal <anuraaga@gmail.com>

If this code is too deep for such comments, feel free to close no worries. Happened to be something mysterious I ran into in first iterations of #892

### CODE REVIEW

File: internal/engine/compiler/engine.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ type (
 		// name is the name the module was instantiated with used for error handling.
 		name string
 
+		// NOTE: The offset of the functions field is referenced from assembly, so
+		// adding any fields here will cause failure at runtime. They should be
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply internal/engine/compiler/engine.go]: Ah that's much more precise, tweaked the comment to reference it

File: internal/engine/compiler/engine.go [hunk 2]
```diff
@@ -39,6 +39,10 @@ type (
 		// name is the name the module was instantiated with used for error handling.
 		name string
 
+		// NOTE: The offset of the functions field is referenced from assembly based on
```
Comment (mathetake) [inline internal/engine/compiler/engine.go:42]: so this is not moduleEngine specific thing but rather general thing about all the structs in this file.. so move this comment to https://github.com/tetratelabs/wazero/blob/main/internal/engine/compiler/engine.go#L25 ?

Comment (mathetake) [reply internal/engine/compiler/engine.go:42]: basically like "be careful when you modify fields in this file!"

Comment (anuraaga) [reply internal/engine/compiler/engine.go:42]: Interesting - I think if the comment isn't exactly where a problematic field would be added, it wouldn't have prevented my lengthyish debugging session 😅 But it makes sense so moved it up

Comment (mathetake) [reply internal/engine/compiler/engine.go:42]: yeah understood - basically we should dup the same comment across all the fields

Comment (anuraaga) [reply internal/engine/compiler/engine.go]: I tried adding some lightweight dup comments to increase the chance of safety without getting overblown, how does it look?

**Review by codefromthecrypt — APPROVED** [review]
I think this makes a lot of sense and former me probably would have noticed this. Thanks!

**Pull request merged**